internal/helper/workpool.go

```diff
@@ -0,0 +1,46 @@
+package helper
+
+import "sync"
+
+type WorkFunc interface {
+	Run()
+}
+
+type GoroutinePool struct {
```

Comment:
This poolcould be part of your service, as it is used for a general purpose, it can be left on service package.